go/pvl/debug.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2016 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package pvl
+
+import (
+	"fmt"
+
+	libkb "github.com/keybase/client/go/libkb"
+	keybase1 "github.com/keybase/client/go/protocol"
+)
+
+func debugWithState(g libkb.ProofContext, state PvlScriptState, format string, arg ...interface{}) {
+	s := fmt.Sprintf(format, arg...)
+	g.GetLog().Debug("PVL @(service:%v script:%v pc:%v) %v",
+		debugServiceToString(state.Service), state.WhichScript, state.PC, s)
+}
+
+func debugWithStateError(g libkb.ProofContext, state PvlScriptState, err libkb.ProofError) {
+	g.GetLog().Debug("PVL @(service:%v script:%v pc:%v) Error code=%v: %v",
+		debugServiceToString(state.Service), state.WhichScript, state.PC, err.GetProofStatus(), err.GetDesc())
+}
+
+func debugWithPosition(g libkb.ProofContext, service keybase1.ProofType, whichscript int, pc int, format string, arg ...interface{}) {
+	s := fmt.Sprintf(format, arg...)
+	g.GetLog().Debug("PVL @(service:%v script:%v pc:%v) %v",
+		debugServiceToString(service), whichscript, pc, s)
+}
+
+func debug(g libkb.ProofContext, format string, arg ...interface{}) {
+	s := fmt.Sprintf(format, arg...)
+	g.GetLog().Debug("PVL %v", s)
```

Comment:
I love this tooling for debugging -- however, the problem is that all file/line output will just say `debug.go:26`. There is a mechanism to change the stack frame we're looking relative to, you might want to plumb that through.